Automattic/jetpack
# [paid-newsletters] visibility selector & wording changes

Fixes #28590 and #28592

## Proposed changes:
* Align newsletter UI menu with stock WordPress UI (see Visibility menu)
* Fix some wording
* Remove number of social followers if 0

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:
* Patch your sandbox
* Open a paid-newsletter-enabled blog (for example https://bestmemberships.wordpress.com/)
* Create a new post and watch the paid newsletter menu

You should see:
 

| Edition | Edition (menu opened) | Pre-publish |
| -- | -- | -- |
|  <img width="988" alt="Screenshot 2023-02-13 at 17 25 40" src="https://user-images.githubusercontent.com/790558/218514835-ddb777ea-7964-435d-b934-5128188aa476.png"> | <img width="991" alt="Screenshot 2023-02-13 at 17 26 03" src="https://user-images.githubusercontent.com/790558/218514878-c0edbf9d-628d-45ad-9529-47ee23af2ed6.png"> | <img width="985" alt="Screenshot 2023-02-13 at 17 26 14" src="https://user-images.githubusercontent.com/790558/218514900-a425063d-5c3b-4634-98ab-ad7e82d128d6.png"> |
